### PR TITLE
[CMDCT-159] Target Populations added to SAR Form Creation

### PR DESCRIPTION
**Pull request by ntsummers1**
### Description

Does this work? Am I crazy? Why not both? You decide.
Prepare your mental fortitude and if you can make it through the madness I wrote today, then Im glad someone was able to.

https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/19439679/782be747-561e-48ed-bf35-1ee17db58a26

### Related ticket(s)

CMDCT-159

---
### How to test

Create a work plan
Add a transition benchmark, if you want,
Create a SAR and see its in the modal!
Edit a SAR on the dashboard and see its in the modal!

### Important updates

---
### Author checklist

- [x] I have performed a self-review of my code
- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [x] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/utils/forms/forms.ts [hunk 1]
```diff
@@ -228,3 +231,80 @@ export const resetClearProp = (fields: (FormField | FormLayoutElement)[]) => {
...
+    return {
+      key: `targetPopulations-${field.id}`,
+      value: field.isRequired
+        ? field.transitionBenchmarks_targetPopulationName
+        : `Other: ${field.transitionBenchmarks_targetPopulationName}`,
+    };
+  });
+};
+
+export const convertChoiceToEntity = (choices: Choice[]) => {
```
Comment (ajaitasaini) [inline services/ui-src/src/utils/forms/forms.ts:248]: non-blocking: would be super helpful for the future if you left a quick comment defining what the difference between the following 2 functions are and when to use each (`convertChoiceToEntity`, and `convertTargetPopulationsFromWPToSAREntity`)

File: services/ui-src/src/utils/forms/forms.ts [hunk 2]
```diff
@@ -228,3 +231,80 @@ export const resetClearProp = (fields: (FormField | FormLayoutElement)[]) => {
...
+      id: field.key,
+      label: field.value,
+      name: field.value,
+      value: field.value,
+    };
+  });
+};
+
+export const convertTargetPopulationsFromWPToSAREntity = (
+  targetPopulations: any[]
```
Comment (ajaitasaini) [inline services/ui-src/src/utils/forms/forms.ts:260]: does this have to be an any? can it be AnyObject[]?

File: services/ui-src/src/components/pages/Dashboard/DashboardTable.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ import {
   ReportType,
   TableContentShape,
 } from "types";
-import { convertDateUtcToEt } from "utils";
+import { convertDateUtcToEt, prettyPrintChoices } from "utils";
```
Comment (keeysnc) [inline services/ui-src/src/components/pages/Dashboard/DashboardTable.tsx:11]: nit: Maybe `formatPrintChoices` might be more descriptive

Comment (keeysnc) [reply services/ui-src/src/components/pages/Dashboard/DashboardTable.tsx:11]: Just noticed Ajaita had a similar comment. lol

**Review by keeysnc — DISMISSED** [review]
This was a lot of refactoring! I only had one small comment but besides that it looks good and works on my end, nice work.

File: services/ui-src/src/utils/other/rendering.ts [hunk 4]
```diff
@@ -14,3 +14,6 @@ export const eligibilityGroup = (entity: EntityShape) => {
   }
   return entity.report_eligibilityGroup[0].value;
 };
+
+export const prettyPrintChoices = (choices: Choice[]) =>
```
[the remark this answers is not in the document]
Comment (ntsummers1) [reply services/ui-src/src/utils/other/rendering.ts:18]: For sure!

File: services/ui-src/src/utils/forms/forms.ts (hunk 1 above)
Comment (ntsummers1) [reply services/ui-src/src/utils/forms/forms.ts:248]: Done :D

File: services/ui-src/src/utils/forms/forms.ts (hunk 2 above)
Comment (ntsummers1) [reply services/ui-src/src/utils/forms/forms.ts:260]: Yes!

**Review by ajaitasaini — DISMISSED** [review]

**Comment by ajaitasaini** [comment]
Thanks for addressing the feedback + adding super descriptive comments - amazing job on this!!

**Review by ajaitasaini — APPROVED** [review]

**Review by keeysnc — APPROVED** [review]

**Pull request merged**